wire-compiler/src/main/java/com/squareup/wire/schema/WireRun.kt

```diff
@@ -271,6 +273,8 @@ data class WireRun(
           module = module,
           profileLoader = schemaLoader,
         )
+        // Ensure the out directory is created.
+        fs.createDirectories(outDirectory)
```

Comment:
This was previously done lazily, right? So if no files were actually generated, it would not end up creating the door output directory. Is this change intentional? Is it not sufficient to lazily create directories right before creating generated files?

Reply:
There's actually a test which I'm trying to keep passing which I think assumes this type of behavior. Let's keep note on this and surface this up with the block folks? I can make the change to that behavior on whatever we decide